database/src/pool/postgres.rs

```diff
@@ -1985,11 +1989,12 @@ where
                         AND target = $2
                         AND benchmark_set = $3
                     ORDER BY
-                        -- Prefer in-progress jobs that have not been finished previously, so that
-                        -- we can finish them.
+                        -- Prefer in-progress jobs that have not been finished
+                        -- previously, so that we can finish them. Also prefer
+                        -- mandatory jobs over optional ones.
                         CASE
-                            WHEN status = $5 THEN 0
-                            WHEN status = $1 THEN 1
+                            WHEN (status = $5 AND is_optional = FALSE) THEN 0
```

Comment:
I don't think that we have to prefer non-optional jobs. It shouldn't even be an issue in practice, as we expect to use optional jobs for different targets/collectors, so a single collector should ideally only ever do required or only ever do optional jobs, and not mix-and-match.